### PR TITLE
patch st2-run-pack-tests to work with python3

### PR DESCRIPTION
**Pull request by guzzijones**
allows st2-run-pack-tests to work with python3 on systems that have python3 installed as a separate binary.

I probably need to find and or create an integration test for this.

### CODE REVIEW

**Comment by guzzijones** [comment]
I see 2 places where this script is called:
https://github.com/StackStorm/st2/blob/b54262709f02b17acb716ed12efc651c71e1405d/Makefile
https://github.com/StackStorm/st2/blob/2b7c4005243fc268497b6780333241e9fc712f82/tox.ini


Adding a test of the -3 will require a git clone of the repo and setting the ST2_REPO_PATH and then running .

**Comment by armab** [comment]
@nzlosh Can you please help evaluating UX functionality/changes in this PR and review it?

**Comment by nzlosh** [comment]
Can I have some more context about the problem this PR is addressing?  On Ubuntu 18.04 the `st2-run-pack-tests` detects and uses python3.6 to test the pack without any problem.  Running the script with `-3` returns `must set ST2_REPO_PATH variable`

**Comment by guzzijones** [comment]
The centos7 install uses python2.7. Adding this switch will tell the pack
test script to NOT look for the st2 packages in the installed dir at
/opt/stackstorm.
Yes on ubuntu you do not need this. The ubunu install uses python3.

Furthermore, to test this correctly we need to run this script with the new
-3 flag on a centos7 box. You cant simply run tests on en empty box. And we
cant * that i know of * use the tox.ini because that wont have the versions
conflict with python.

To reproduce the bug this fixes try to run a python3 action script though
the original on Stackstorm centos7 installed box. You will see it try to
load up python2.7 libs into its pythonpath. And the test will fail.

In the end this is dead end code. But i have to use this version on our st2
installed testing box.
I submitted it to help others who run onto the same issue with testing on a
stackstorm centos7 box.

**Comment by guzzijones** [comment]
Essentially this flag says, "python is 2.7 and python3 is 3 on the system.

**Comment by armab** [comment]
Here is the complete `py2` vs `py3` OS support matrix:

Python 3:
- EL8
- Ubuntu Bionic

Python 2:
- EL6 (will be deprecated in st2 v3.3)
- EL7
- Ubuntu Xenial

**Comment by armab** [comment]
@nzlosh did @guzzijones answer your question?
Any further PR review/feedback based on that context?

File: st2common/bin/st2-run-pack-tests [hunk 1]
```diff
@@ -104,6 +107,14 @@ while getopts ":p:f:xjvct" o; do
         t)
             ENABLE_TIMING=true
             ;;
+        3)
+            PYTHON_3=true
+            if [ -z "$ST2_REPO_PATH" ]; then
+                echo "must set ST2_REPO_PATH variable"
```
Comment (nzlosh) [inline st2common/bin/st2-run-pack-tests:113]: A clearer error message explaining what the `ST2_REPO_PATH` variable is used for will help people set the correct value. e.g. `Environment variable ST2_REPO_PATH must be set to your st2 installation or st2 git repository for pack tests to function correctly.`

File: st2common/bin/st2-run-pack-tests [hunk 2]
```diff
@@ -104,6 +107,14 @@ while getopts ":p:f:xjvct" o; do
         t)
             ENABLE_TIMING=true
             ;;
+        3)
+            PYTHON_3=true
+            if [ -z "$ST2_REPO_PATH" ]; then
```
[the remark this answers is not in the document]
Comment (guzzijones) [reply st2common/bin/st2-run-pack-tests:112]: This is the most important part. This script will look for st2 libs in \opt\stsckstorm if i dont set this. On centos7 those are python 2.7. Maybe what we do here is check the system python version vs. The version passed in this flag. If they doffer then force this env var to be set.

**Comment by guzzijones** [comment]
Just a heads up:
If this gets milestone I will spend some time on it.  Currently I am working on other PRs.

**Comment by armab** [comment]
@guzzijones Yes, we'll be good to accept this enhancement into st2 core, once it's polished.

Thanks @nzlosh for helping with the review to get it closer for the broader use.

**Comment by armab** [comment]
@guzzijones any update on driving this PR to completion?

**Comment by cognifloyd** [comment]
We do not support python2 now, so is this needed anymore?

**Comment by amanda11** [comment]
> We do not support python2 now, so is this needed anymore?

I don't think so, the last two ST2 releases (3.4 and 3.5) do not support python2.

**Comment by cognifloyd** [comment]
I'm going to close this then. If we still need it for some reason, feel free to reopen.

**Comment by armab** [comment]
Closing due to py2 is not relevant anymore.
Thanks @cognifloyd for spotting this!

**Pull request closed without merging**